android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -275,6 +280,49 @@ public void updateTabTitle(int index)
 	}
 
 	@SuppressLint("RestrictedApi")
+	@Override
+	public void updateBadge(int index)
+	{
+		if ((index < 0) || (index >= this.tabs.size())) {
+			return;
+		}
+
+		TiViewProxy tabProxy = this.tabs.get(index).getProxy();
+		if (tabProxy == null) {
+			return;
+		}
+
+		int menuItemId = this.mBottomNavigationView.getMenu().getItem(index).getItemId();
+		if (tabProxy.getProperty(TiC.PROPERTY_BADGE) != null) {
+			BadgeDrawable badgeDrawable = this.mBottomNavigationView.getOrCreateBadge(menuItemId);
+			badgeDrawable.setVisible(true);
+			badgeDrawable.setNumber(TiConvert.toInt(tabProxy.getProperty(TiC.PROPERTY_BADGE), 0));
+		} else {
+			this.mBottomNavigationView.removeBadge(menuItemId);
```

Comment:
Perhaps we should do a `setVisible(false)` instead of doing a `removeBadge()`. This way, we'll be able to set the "badgeColor" property while the "badge" property is `null`. _(Alternatively, you can set the badge color every time you set the badge number.)_

So, this is to handle the case where devs initialize the "badgeColor" first and set the badge value later, such as after receiving the results of an HTTP request.
```javascript
function createTab(title) {
	var window = Ti.UI.createWindow({ title: title });
	window.add(Ti.UI.createLabel({ text: title + " View" }));
	var tab = Ti.UI.createTab({
		title: title,
		icon: "/assets/images/tab1.png",
		badgeColor: "purple",
		window: window,
	});
	return tab;
}

var tabGroupSettings = {
	tabs: [createTab("Tab 1"), createTab("Tab 2"), createTab("Tab 3")],
};
if (OS_ANDROID) {
	tabGroupSettings.shiftMode = false;
	tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;
}
var tabGroup = Ti.UI.createTabGroup(tabGroupSettings);
tabGroup.addEventListener("open", function() {
	tabGroup.tabs[1].badge = "42";
});
tabGroup.open();
```

Reply:
works fine with hiding/showing it